### PR TITLE
feature: add support for database upload from the working directory

### LINKED ISSUE #195: Add support for uploading database dump without using a docker container
Currently, in order to upload a database to the project the process it to use:

```
divio project push db
```
This actually spins up a docker container for the project, gets the db dump, compresses it, uploads and if all goes well clear the dump files from the working directory..

It would help if the cli support uploading db dump directly without having to setup the project.

With the support for command line flags to specify website id it is possible to do this.

### PR DESCRIPTION
**Pull request by vinitkumar**
This PR adds the support for uploading the database dump directly from
the working directory. The user needs to provide the website id where
the database needs to be uploaded and a valid dump file that could be
restored.

In order to make sure the db dump is compatible, create a dump using the
following command:

```
pg_dump --no-privileges --no-acl --no-owner -d cms > cms_dump.sql
```
Now, in order to upload dump, run this command from the directory where
the file exists.

```
 divio project --id=37457 push db --dumpfile=cms_dump.sql
```

This will upload the file after compressing it and restore it in the
same way it does while uploading db from docker container for the
project.

- Github Issue: fixes #195

Signed-off-by: Vinit Kumar <vinit.kumar@socialschools.nl>

### CODE REVIEW

**Comment by vinitkumar** [comment]
@Chive please review this one.

**Comment by vinitkumar** [comment]
@Chive I had removed the asserts from this PR. Also, archive_wd is the directory where the dump file exists as I didn't wanted it to do anything with the project_home.

File: divio_cli/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '3.3.0'
+__version__ = '3.3.1'
```
Comment (Chive) [inline divio_cli/__init__.py]: Same as before, please don't bump the version yet - we'll do that in a separate commit.

File: divio_cli/cli.py [hunk 2]
```diff
@@ -429,6 +429,24 @@ def push_db(obj, stage, noinput):
...
+    test (default) or live
+    """
+    website_id = obj.project["id"]
+    if not noinput:
+        click.secho(messages.PUSH_DB_WARNING.format(stage=stage), fg='red')
+        if not click.confirm('\nAre you sure you want to continue?'):
+            return
+    localdev.push_local_db(obj.client, stage, dump_filename, website_id)
+
+
```
Comment (Chive) [inline divio_cli/cli.py]: Waiting for your change as discussed in https://github.com/vinitkumar/divio-cli/pull/1#discussion_r126087402

File: divio_cli/localdev/main.py [hunk 3]
```diff
@@ -571,6 +599,57 @@ def push_db(client, stage):
     click.echo(' [{}s]'.format(int(time() - start_time)))
 
 
+def push_local_db(client, stage, dump_filename, website_id):
+    if not os.path.exists(dump_filename):
+        click.secho("The dumpfile {} doesn't seem to exists in the current directory".format(dump_filename))
```
Comment (Chive) [inline divio_cli/localdev/main.py]: doesn't seem to **exist**

File: divio_cli/__init__.py (hunk 1 above)
Comment (vinitkumar) [inline divio_cli/__init__.py]: @Chive I will remove this. Thanks for pointing out.

File: divio_cli/cli.py (hunk 2 above)
Comment (vinitkumar) [inline divio_cli/cli.py]: @Chive Will update the code to ensure it is hidden behind a flag as you suggested

File: divio_cli/localdev/main.py (hunk 3 above)
Comment (vinitkumar) [inline divio_cli/localdev/main.py]: @Chive Thanks for pointing out the grammar mistake again.

**Comment by vinitkumar** [comment]
@Chive I have made the suggested changes, the upload is behind a flag now. It requires a project id to be passed as well as the file needs to be in the current directory where the command is executed. Hopefully, if it good enough to be merged. I have already tested all the edge case. Please check it once at your end too. If you need any changes, let me know!

Thanks!

File: divio_cli/cli.py [hunk 4]
```diff
@@ -415,19 +415,34 @@ def project_push():
 
 @project_push.command(name='db')
 @click.argument('stage', default='test')
-@click.option('--noinput', is_flag=True, default=False, help="Don't ask for confirmation")
+@click.option(
```
Comment (Chive) [inline divio_cli/cli.py]: you can use Click's builtin file type with checking paths ([see here](https://github.com/divio/divio-cli/blob/master/divio_cli/cli.py#L458))

Comment (vinitkumar) [inline divio_cli/cli.py]: @Chive This is done as well.
